### DISCUSSION TITLE
Export array into csv file using Mirth Connect

### DISCUSSION THREAD

**Discussion by ShoaibNaseerAppifyTech** [nextgenhealthcare/connect#4596] [Q&A]
I am importing a csv file by the following line:
var CSV = connectorMessage.getEncodedData();

I am using JavascriptWriter for this problem.
My objective is to eliminate the empty column from this and export it. So I made an array which has no empty column.
 ["1","ali","12/10/2000","2","usama","12/4/2006","3","qasim","12/10/2003","4","taha","12/10/1995","5","","12/10/1992"]
I want to export this array into csv when file move to particular folder.
Kindly help me on this query.
Thanks

**Comment by pacmano1** [comment]
I would go over to the mirth forums in this case and take a look at the many posts about CSVs.    

Also, take a look at the forum etiquette https://forums.mirthproject.io/articles/177082-forum-etiquette, which applies here as well.

In particular, at least post what you tried.

**Reply by ShoaibNaseerAppifyTech** [reply]
But I did not find anything related to this:
https://forums.mirthproject.io/forum/mirth-connect/support/7271-parsing-csv
This forum helps to get the data from the csv and I use it but how to save the data from array into CSV using JavaScript Writer in Destination Tab. The searches which I found is mostly related to XML to CSV or JSON to CSV. Some of the examples are below:
1. https://forums.mirthproject.io/forum/mirth-connect/support/16869-?t=217724
2. https://forums.mirthproject.io/forum/mirth-connect/support/12916-database-table-to-csv-file
In the  second link I use channelMap but its not working nor giving any other error. And also it is not present on right dialogue box option. 
![2](https://user-images.githubusercontent.com/79902431/122019461-3aa09100-cddd-11eb-8215-9453b0ee9d09.PNG)

![1](https://user-images.githubusercontent.com/79902431/122019340-1ba1ff00-cddd-11eb-8378-0eb820db0d21.PNG)

**Comment by tonygermano** [comment]
When you use the delimited data type, your csv is represented as XML in the transformer. You can do CSV to different CSV. In your case, probably the easiest thing to do would be to delete the column you don't need from the row in the transformer, and let mirth handle the serialization.

There is no way to automatically convert an array of strings to a csv (which itself is just a string.) If you need to do it that way, you'll have to build the csv string yourself from your array.

**Reply by ShoaibNaseerAppifyTech** [reply]
How you check that which column is empty and which is non empty programmatically in transformer ?

**Reply by tonygermano** [reply]
You should edit your original post to provide more information about your problem with sample input and desired output. You never mentioned needing to detect which column is empty, and I assumed you needed to always delete the same column for every row in the input to meet some desired output format.

**Reply by tom08zehn** [reply]
Your array looks wrong, the result would be a "flat" CSV with just one row.

You need an array of arrays 😀 to represent a CSV like this (simplified):

```js
// create table using arrays
var row1 = ["colA1", "colB1"];
var row2 = ["colA2", "colB2"];
var table = [row1, row2];
// stringify to CSV
var csv = [];
// step 1: stringify cols, basically concat all cols of a row with comma
table.forEach(row => {
    csv.push(row.join(","));
});
// step 2: stringify rows, concat with linebreak
csv = csv.join("\n");
```
Hint: this is a simplified version not supporting all CSV features according to the RFC definition, e.g. the result will be wrong if a value contains a comma character.

**Reply by ShoaibNaseerAppifyTech** [reply]
The problem is solved by using the same array by made another destination in the same channel

**Comment by ShoaibNaseerAppifyTech** [comment]
I create another destination in which the File write is used and get the data using the channelMap
